Feature: Input form for institutions

mukkig opened the pull request:
## 📝 What was changed?
This PR introduces a new view that allows users to create new institution records. A corresponding controller has also been defined. Additionally, a test has been added to cover the new input form functionality.

The README has been updated accordingly.

## 🔗 Issue
Closes #151 

## 🧪 Definition of Done erfüllt?

- [x] User story fulfilled
- [x] Documentation updated
- [x] New code is covered by unit tests
- [x] Unit tests locally pass
- [x] Manually tested
- [x] Browser compatible
- [x] Mobile compatible

## 🚀 Laravel-specific

- [ ] Migrations added/changed
- [ ] Models added/changed
- [x] Views added/changed
- [x] Routes registered
- [ ] Seeder added/changed
- [ ] Config changed
- [ ] Composer dependencies updated
- [ ] Artisan Commands added/changed

## 📷 Screenshots
<!-- If UI changed -->

## 💬 Notes
I'm using the InstitutionFactory, which can generate records that sometimes contain invalid data (due to its defined logic). Therefore, the test may fail occasionally. I've noticed this when running multiple iterations. Please keep this in mind and review the factory logic to understand the cause.

McNamara84 commented on tests/Feature/InputFormInstitutionTest.php:
```diff
@@ -0,0 +1,135 @@
+     * does not get stored in the database and redirects with appropriate errors.
+     */
+
+        public function test_store_does_not_create_institution_because_of_invalid_value_in_record_array_and_redirects(): void
```
Indent

mukkig replied:
done

McNamara84 reviewed (changes requested):
Thanks for this post. Being able to manually add institutions is an important feature for our stakeholders.

The front end works fine for me. I would just take a look at the dropdown for the type. You have set a long text as the default option, which makes this dropdown wider than it is. I would suggest moving this text below the dropdown as help text and changing the default option to “Bitte auswählen”

Saving works perfectly. I was able to write several institutions to the database without any errors.

I like the code style. I only found one small problem with an indentation, which I would ask you to fix.

The tests run without any problems, but there are issues if you have already created n institutions yourself to run the test. Then the following test case fails:
<img width="704" height="403" alt="Image" src="https://github.com/user-attachments/assets/c15adc24-83b8-424a-bdbc-e399c8b0e687" />
This is a problem because the tests are also performed on our server via the CD pipeline. Automated deployment would throw an error as soon as the prod DB was filled with more data and you moved this contribution to main. There are better ways to create a test here. Please take a look at it.

Overall this will be a great contribution. I really appreciate that you are making so many extensive contributions to this project. And this will take the project another big step forward. Thanks!

mukkig commented:
hi @McNamara84 I have updated the test file and implemented your recommendations regarding the input form. Thank you very much for your valuable feedback!

The test no longer uses the InstitutionFactory. Instead, I am using Faker with appropriate methods to generate values tailored to the defined test scenarios.
I hope this makes the test more robust now.

I look forward to your feedback.
